src/Symfony/Component/AstGenerator/AstGeneratorInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AstGenerator;
+
+/**
+ * An AstGeneratorInterface is a contract to transform an object into an AST.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+interface AstGeneratorInterface
+{
+    /**
+     * Generate an object into an AST given a specific context.
+     *
+     * @param mixed $object  Object to generate AST from
+     * @param array $context Context for the generator
+     *
+     * @return \PhpParser\Node[] An array of statements (AST Node)
+     */
+    public function generate($object, array $context = array());
+
+    /**
+     * Check whether the given object is supported for generation by this generator.
+     *
+     * @param mixed $object Object to generate AST from
+     *
+     * @return bool
+     */
+    public function supportsGeneration($object);
```

Comment:
This method makes me wonder: how it is possible that we don't exactly know if our generator is allowed to work with the given object? Does it even make sense to "try" generate AST tree of "random" object? Does it make sense to replace `UrlGeneratorGenerator` with `PhpMatcherGenerator` somewhere? I don't think so, they have very different semantic sense and they probably should have different (specific) interfaces: `interface UrlGeneratorGenerator`, `interface PhpMatcherGenerator` if you really want to have multiple implementations. 

Once interface becomes vague, it turns into wrong abstraction, because 
> The purpose of abstraction is not to be vague, but to create a new semantic level in which one can be absolutely precise

Also, it feels like "Tell, don't ask" violation.